config/config_test.go

```diff
@@ -2437,9 +2440,9 @@ func TestScrapeConfigDisableCompression(t *testing.T) {
 
 func TestScrapeConfigNameValidationSettings(t *testing.T) {
 	model.NameValidationScheme = model.UTF8Validation
-	defer func() {
+	defer t.Cleanup(func() {
 		model.NameValidationScheme = model.LegacyValidation
-	}()
+	})
```

Comment:
Why do you make this change? It's at least wrong to defer a `t.Cleanup` call.

Reply:
Apologies, I made the change in the assumption of the Cleanup needs to be deferred until the test/subtest completes, but `t.Cleanup` doesn't need to defer.